### PR TITLE
GH1044: Fix tests code analyze violations

### LINKED ISSUE #1044: Fix test project Cake code standard violations

### PR DESCRIPTION
**Pull request by devlead**
Fixes #1044

### CODE REVIEW

**Comment by gep13** [comment]
@devlead is there an issue for this? :wink:

**Comment by devlead** [comment]
@gep13 No... ;)

File: src/Cake.Testing/Extensions/FakeFileExtensions.cs [hunk 1]
```diff
@@ -18,8 +20,19 @@ public static class FakeFileExtensions
...
         /// <param name="content">The content.</param>
         /// <returns>The same <see cref="FakeFile"/> instance so that multiple calls can be chained.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times")]
         public static FakeFile SetContent(this FakeFile file, string content)
         {
+            if (file == null)
+            {
+                throw new ArgumentNullException("file");
+            }
+
```
Comment (patriksvensson) [inline src/Cake.Testing/Extensions/FakeFileExtensions.cs]: No extra line here.

File: src/Cake.Testing/Extensions/FakeFileExtensions.cs [hunk 2]
```diff
@@ -35,8 +48,14 @@ public static FakeFile SetContent(this FakeFile file, string content)
...
         /// <param name="file">The file.</param>
         /// <returns>The text content of the file.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times")]
         public static string GetTextContent(this FakeFile file)
         {
+            if (file == null)
+            {
+                throw new ArgumentNullException("file");
+            }
+
```
Comment (patriksvensson) [inline src/Cake.Testing/Extensions/FakeFileExtensions.cs]: No extra line here.

File: src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs [hunk 3]
```diff
@@ -19,6 +21,16 @@ public static class FakeFileSystemExtensions
         /// <param name="path">The path.</param>
         public static void EnsureFileDoNotExist(this FakeFileSystem fileSystem, FilePath path)
         {
+            if (fileSystem == null)
+            {
+                throw new ArgumentNullException("fileSystem");
+            }
+
```
Comment (patriksvensson) [inline src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs]: No extra line here.

File: src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs [hunk 4]
```diff
@@ -34,6 +46,16 @@ public static void EnsureFileDoNotExist(this FakeFileSystem fileSystem, FilePath
         /// <returns>The same <see cref="FakeFile"/> instance so that multiple calls can be chained.</returns>
         public static FakeFile CreateFile(this FakeFileSystem fileSystem, FilePath path)
         {
+            if (fileSystem == null)
+            {
+                throw new ArgumentNullException("fileSystem");
+            }
+
```
Comment (patriksvensson) [inline src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs]: No extra line here.

File: src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs [hunk 5]
```diff
@@ -54,6 +76,21 @@ public static FakeFile CreateFile(this FakeFileSystem fileSystem, FilePath path)
         /// <returns>The same <see cref="FakeFile"/> instance so that multiple calls can be chained.</returns>
         public static FakeFile CreateFile(this FakeFileSystem fileSystem, FilePath path, byte[] contentsBytes)
         {
+            if (fileSystem == null)
+            {
+                throw new ArgumentNullException("fileSystem");
+            }
+
```
Comment (patriksvensson) [inline src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs]: No extra line here.

File: src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs [hunk 6]
```diff
@@ -54,6 +76,21 @@ public static FakeFile CreateFile(this FakeFileSystem fileSystem, FilePath path)
...
+            if (path == null)
+            {
+                throw new ArgumentNullException("path");
+            }
+
```
Comment (patriksvensson) [inline src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs]: No extra line here.

File: src/Cake.Testing/Fixtures/ToolFixtureResult.cs [hunk 7]
```diff
@@ -48,6 +50,16 @@ public string Args
         /// <param name="process">The process settings.</param>
         public ToolFixtureResult(FilePath path, ProcessSettings process)
         {
+            if (path == null)
+            {
+                throw new ArgumentNullException("path");
+            }
+
```
Comment (patriksvensson) [inline src/Cake.Testing/Fixtures/ToolFixtureResult.cs]: No extra line here.

File: src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs (hunk 4 above)
Comment (patriksvensson) [inline src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs]: No extra line here.

File: src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs (hunk 5 above)
Comment (patriksvensson) [inline src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs]: No extra line here.

File: src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs (hunk 6 above)
Comment (patriksvensson) [inline src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs]: No extra line here.

File: src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs [hunk 8]
```diff
@@ -79,6 +116,16 @@ public static FakeFile CreateFile(this FakeFileSystem fileSystem, FilePath path,
         /// <returns>The same <see cref="FakeDirectory"/> instance so that multiple calls can be chained.</returns>
         public static FakeDirectory CreateDirectory(this FakeFileSystem fileSystem, DirectoryPath path)
         {
+            if (fileSystem == null)
+            {
+                throw new ArgumentNullException("fileSystem");
+            }
+
```
Comment (patriksvensson) [inline src/Cake.Testing/Extensions/FakeFileSystemExtensions.cs]: No extra line here.

**Comment by devlead** [comment]
@patriksvensson think excessive line breaks should be addressed now.

**Comment by devlead** [comment]
@gep13 issue #1044 created & rebased

**Pull request merged**